scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -578,12 +569,20 @@ def __emit_report_if_necessary(self, current_time=None):
         if self.__last_report_time is None or current_time > self.__last_report_time + 300:
             self.__last_report_time = current_time
 
+            global_log.info('echee: __emit_report_if_necessary : emitting')
             warm_attempts_info = ''
             for category in ['total', 'success', 'temp_error', 'perm_error', 'unknown_error', 'unhandled_error']:
                 current_amount = self.__warming_attempts.get(category, 0)
+                global_log.info('echee: category={}, current_amount={}'.format(category, current_amount))
                 previous_amount = self.__last_reported_warming_attempts.get(category, 0)
+                global_log.info('echee: category={}, current_amount={}'.format(category, previous_amount))
                 warm_attempts_info += '%s=%d(delta=%d) ' % (category, current_amount, current_amount - previous_amount)
-            self.__logger.info('controlled_cache_warmer pending_warming=%d %blacklisted=%d %s',
+                global_log.info('echee: category={}, current_amount={}'.format(category, previous_amount))
+            self.__logger.info('controlled_cache_warmer pending_warming=%d blacklisted=%d %s',
```

Comment:
Notice the very subtle bug in Steven's "blacklisted" . Changing exception handlers to log using `logger.exception` instead of `logger.error` were instrumental in finding this bug. 

We really should use logger.exception whenever possible and avoid logger.error unless we know for sure we want to suppress the stack. (alternatively include the stack via traceback)

Reply:
Agree we should discuss.  I think this  issue applies everywhere we log traceback.format_exc() or traceback.format_stack() as well so we need consistency